### PR TITLE
add TreeShrink

### PR DESCRIPTION
**Pull request by tomkinsc**
This adds a recipe for [TreeShrink](https://github.com/uym2/TreeShrink), a tool for detecting (and removing) abnormally long branches in one or more phylogenetic trees. 

The associated TreeShrink paper:
> Mai, Uyen, and Siavash Mirarab. 2018. “TreeShrink: Fast and Accurate Detection of Outlier Long Branches in Collections of Phylogenetic Trees.” BMC Genomics 19 (S5): 272. https://doi.org/10.1186/s12864-018-4620-2

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/treeshrink/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  number: 0
+  noarch: generic
+
+source:
+  url: https://github.com/uym2/TreeShrink/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 460894699250b7fc1a7096d84bebb0b52b5551612e9faa283c600a15d3f756a3
+
+requirements:
+  host:
+    - python >=3.6,<=3.9.15
```
Comment (bgruening) [inline recipes/treeshrink/meta.yaml:17]: This pin looks strange. Is that really needed?

File: recipes/treeshrink/build.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+R CMD INSTALL -l Rlib dependencies/BMS_0.3.3.tar.gz
```
Comment (bgruening) [inline recipes/treeshrink/build.sh:3]: We should create an own pakchage for that.

File: recipes/treeshrink/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+about:
+  home: https://github.com/uym2/TreeShrink
+  license_file: LICENSE
+  license: GPL-3
+  license_family: GPL
+  summary: an algorithm for detecting (and removing) abnormally long branches in one or more phylogenetic trees
+
+extra:
+  recipe-maintainer:
+    - uym2 # (this recipe based on one available here: https://github.com//TreeShrink/tree/master/conda_package/treeshrink)
```
Comment (bgruening) [inline recipes/treeshrink/meta.yaml:40]: Does @uym2 agrees to be maintainer?

Comment (uym2) [reply recipes/treeshrink/meta.yaml:40]: Yes, I am.

File: recipes/treeshrink/meta.yaml (hunk 1 above)
Comment (tomkinsc) [reply recipes/treeshrink/meta.yaml]: Yes, at least until treeshrink is updated upstream; it makes use of `collections.MutableMapping` which is unavailable in Python >=3.10.

**Comment by tomkinsc** [comment]
@BiocondaBot please add label

**Comment by daler** [comment]
@BiocondaBot please update

**Review by daler — APPROVED** [review]

**Comment by daler** [comment]
@BiocondaBot please merge

**Pull request merged**